### PR TITLE
feat(runtime): log formats per deployment environment

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Detect deployment environment according to environment variables (`RISINGWAVE_CI`, `RISINGWAVE_CLOUD`) to use different log formats.

- Local, pretty: human-readable, with tracing spans:
```
  2023-06-28T17:29:43.234322+08:00 DEBUG risingwave_stream::executor::source::source_executor: take snapshot, actor_id: 50, state: [Nexmark(NexmarkSplit { split_index: 2, split_num: 4, start_offset: Some(311298) })]
    at src/stream/src/executor/source/source_executor.rs:285
    in risingwave_stream::executor::wrapper::trace::executor with otel.name: "SourceExecutor 32000027B5 (actor 50, operator 10165)", actor_id: 50
    in risingwave_stream::executor::wrapper::trace::executor with otel.name: "RowIdGenExecutor 3200000001 (actor 50, operator 1)", actor_id: 50
    in risingwave_stream::executor::wrapper::trace::executor with otel.name: "ProjectExecutor 3200000002 (actor 50, operator 2)", actor_id: 50
    in risingwave_stream::executor::actor::actor with otel.name: "Actor 50", actor_id: 50, prev_epoch: 4634010007830528, curr_epoch: 4634010073563136
```
- Cloud, JSON: machine-readable, with tracing spans:
```json
{
    "timestamp": "2023-06-28T17:17:35.404904+08:00",
    "level": "DEBUG",
    "fields": {
        "message": "take snapshot",
        "actor_id": 42,
        "state": "[Nexmark(NexmarkSplit { split_index: 2, split_num: 4, start_offset: Some(315394) })]"
    },
    "target": "risingwave_stream::executor::source::source_executor",
    "span": {
        "actor_id": 42,
        "otel.name": "SourceExecutor 2A000027B5 (actor 42, operator 10165)",
        "name": "executor"
    },
    "spans": [
        {
            "actor_id": 42,
            "curr_epoch": 4633962371809280,
            "otel.name": "Actor 42",
            "prev_epoch": 4633962306142208,
            "name": "actor"
        },
        {
            "actor_id": 42,
            "otel.name": "ProjectExecutor 2A00000002 (actor 42, operator 2)",
            "name": "executor"
        },
        {
            "actor_id": 42,
            "otel.name": "RowIdGenExecutor 2A00000001 (actor 42, operator 1)",
            "name": "executor"
        },
        {
            "actor_id": 42,
            "otel.name": "SourceExecutor 2A000027B5 (actor 42, operator 10165)",
            "name": "executor"
        }
    ]
}
```

- CI, compact: human-readable, without tracing spans:
```
2023-06-28T17:32:48.516494+08:00 DEBUG risingwave_stream::executor::source::source_executor: take snapshot actor_id=41 state=[Nexmark(NexmarkSplit { split_index: 3, split_num: 4, start_offset: Some(606211) })]
```

#10000 #10426

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/utils/runtime/src/lib.rs [hunk 1]
```diff
@@ -194,18 +198,20 @@ pub fn init_risingwave_logger(settings: LoggerSettings, registry: prometheus::Re
...
-        } else {
-            fmt_layer.boxed()
+
+        let fmt_layer = match deployment {
+            Deployment::Ci => fmt_layer
+                .compact()
+                .with_filter(FilterFn::new(|metadata| metadata.is_event())) // filter-out all span-related info
+                .boxed(),
+            Deployment::Cloud => fmt_layer.json().boxed(),
+            Deployment::Other => fmt_layer.pretty().boxed(),
```
Comment (xxchan) [inline src/utils/runtime/src/lib.rs:210]: For local, I'm not sure if `pretty` is too verbose. 🤪

For cloud, is it the best practice to log JSON to stdout? :eyes: I imagined it's sth like writing to some log collector service directly.

Comment (BugenZhao) [reply src/utils/runtime/src/lib.rs:210]: It seems scraping lines from log files and then parsing them with JSON is somehow a common practice in Loki.

- https://grafana.com/docs/loki/latest/clients/promtail/#log-file-discovery
- https://grafana.com/docs/loki/latest/clients/promtail/stages/json/

File: src/stream/src/executor/actor.rs [hunk 2]
```diff
@@ -164,9 +164,17 @@ where
...
         let span_name = format!("Actor {id}");
 
-        let new_span =
-            || tracing::info_span!(parent: None, "actor", "otel.name" = span_name, actor_id = id);
-        let mut span = new_span();
+        let new_span = |epoch: Option<EpochPair>| {
+            tracing::info_span!(
+                parent: None,
+                "actor",
+                "otel.name" = span_name,
```
Comment (TennyZhuang) [inline src/stream/src/executor/actor.rs:171]: What's otel?

**Review by TennyZhuang — APPROVED** [review]

File: src/stream/src/executor/actor.rs (hunk 2 above)
Comment (BugenZhao) [reply src/stream/src/executor/actor.rs:171]: It's a special field for opentelemetry. Will be displayed as the span name.

https://docs.rs/tracing-opentelemetry/latest/tracing_opentelemetry/#special-fields

**Review by Nebulazhang — COMMENTED** [review]
LGTM

**Pull request merged**